### DISCUSSION TITLE
Implementing AudioAPI for Xash3D FWGS engine to add OpenAL Soft/Steam Audio support

### DISCUSSION THREAD

**Discussion by SNMetamorph** [LAGonauta/MetaAudio#41] [Ideas]
Can you help with that in future?

**Comment by ThreeDeeJay** [comment]
How would it work? Would AudioAPI passthrough the sounds with 3D coordinates for OpenAL Soft/Steam Audio to handle the 3D audio mixing? Because if not, it could end up in a situation like Source engine games on Linux, which apparently just use SDL for actual mixing and just pass the stereo mix to OpenAL Soft , which means it would only be able to virtualize the stereo mix if anything, so no true, positional 3D audio or even proper surround sound which is one of the main reasons to use OALS/SA
Either way, I should point out that I'm not really qualified for the programming part, but if LAGonauta isn't around perhaps @hzqst or someone else could take a look 👀

**Reply by SNMetamorph** [reply]
There is no any SoundAPI implementation proposals at the moment. So, feel free to formalizing ideas into some solid concept

**Comment by SNMetamorph** [comment]
As for we have working draft of SoundAPI, so it's not actual anymore.
